File: leetcode/2844-sum-of-squares-of-special-elements/sum-of-squares-of-special-elements.cpp
class Solution {
public:
    int sumOfSquares(vector<int>& nums) {
          int sum_of_squares_of_all_special_elements=0;
        int n=nums.size();
        for(int i=0;i<n;i++){
           
            if(n%(i+1)==0){
                sum_of_squares_of_all_special_elements = sum_of_squares_of_all_special_elements + (nums[i]*nums[i]);
            }
           
        
        }
        return sum_of_squares_of_all_special_elements;
    }
};

File: leetcode/2844-sum-of-squares-of-special-elements/sum-of-squares-of-special-elements.ts
function sumOfSquares(nums: number[]): number {
    let sum = 0;
    const n = nums.length;

    for (let i = 0; i < n; i++) {
        if (n % (i + 1) === 0) {
            sum += nums[i] * nums[i];
        }
    }

    return sum;
}
